### PR TITLE
Implemented cookies for login and authentication

### PR DESCRIPTION
**Pull request by Liljenmaa**
The TSS now uses cookies to handle login information. Included are the following cookies:

username: The username of the currently logged in user
role: The role of the currently logged in user
token: The JWT access token used to authenticate the user to the backend

There are no cookies in the cookies tray if the user is not logged in.
There will be the aforementioned cookies if the user is logged in.
The cookies will be wholly removed if the user logs out.

All functionality relating to localStorage EXCEPT language handling were removed.

The most likely breaking points are the places where these values are used e.g. login, token validation, user privileges for actions...
Tests will most likely be broken in some small way.

NOTE: all references to authorization headers alongside HTTP requests are removed, as the backend now checks the validity of the token through the token in the cookies.
ANOTHER NOTE: the token cookie will be HttpOnly for security reasons, and will NOT be available on the clientside by JS. Other cookies can freely be handled using the techniques presented in the pull request. All the handling of the token cookie will be performed serverside.

### CODE REVIEW

File: front/src/edittracks/tracks.js [hunk 1]
```diff
@@ -238,14 +236,7 @@ const TrackCRUD = () => {
 
   const partialFetch = lodash.partial(fetch, '/api/track'); // eslint-disable-line
 
-  const token = localStorage.getItem('token');
-
-  const opts = {
-    headers: {
-      Authorization: `Bearer ${token}`,
-    },
-  };
```
Comment (TuoLuo994) [inline front/src/edittracks/tracks.js:247]: opts has to be removed from rows 137, 192 ( not defined  no-undef )

**Comment by TuoLuo994** [comment]
Can't log in. error below:

`[0] (node:479) UnhandledPromiseRejectionWarning: JsonWebTokenError: jwt must be provided                   [0]     at Object.module.exports [as verify] (/home/tuoluo/tss/back/node_modules/jsonwebtoken/verify.js:53:17)                                                                                                        [0]     at checkJwt (/home/tuoluo/tss/back/middlewares/jwt.js:11:25)                                       [0]     at Layer.handle [as handle_request] (/home/tuoluo/tss/back/node_modules/express/lib/router/layer.js:95:5)                                                                                                     [0]     at next (/home/tuoluo/tss/back/node_modules/express/lib/router/route.js:137:13)                    [0]     at Route.dispatch (/home/tuoluo/tss/back/node_modules/express/lib/router/route.js:112:3)           [0]     at Layer.handle [as handle_request] (/home/tuoluo/tss/back/node_modules/express/lib/router/layer.js:95:5)                                                                                                     [0]     at /home/tuoluo/tss/back/node_modules/express/lib/router/index.js:281:22                           [0]     at Function.process_params (/home/tuoluo/tss/back/node_modules/express/lib/router/index.js:335:12) [0]     at next (/home/tuoluo/tss/back/node_modules/express/lib/router/index.js:275:10)                    [0]     at Function.handle (/home/tuoluo/tss/back/node_modules/express/lib/router/index.js:174:3)          [0] (node:479) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)                                                                       ^`

**Comment by TuoLuo994** [comment]
Above error happens on Chrome and Edge, not on Firefox.

**Comment by Liljenmaa** [comment]
If connection is unsecure (such as localhost:3000), cookies will be delivered without secure status.

File: front/src/edittracks/tracks.js (hunk 1 above)
Comment (Liljenmaa) [reply front/src/edittracks/tracks.js:247]: Removed.

**Comment by TuoLuo994** [comment]
Looks good!

**Pull request merged**